arduino/smartcar/smartcar.ino

```diff
@@ -55,6 +55,71 @@ const int BACKWARD_SPEED_LIMIT = -50;
 const int MAX_STEERING_ANGLE = 60;
 const auto ONE_SECOND = 1000UL;
 
+const int BOX_WIDTH = 70;
+const int BOX_HEIGHT = 35;
+const int PARKING_ROWS = 5;
+const int PARKING_COLS = 3;
+const int ENTRANCE_R = 4;
+const int ENTRANCE_C = 1;
+
+/*--- INTERNAL MAP SETUP---*/
+
+enum BoxType{Path, Unoccupied, Occupied, Start, NaP};
+
+class GridBox{
+  public:
+    bool hasCar;
+    BoxType type;
+    GridBox(BoxType boxType){
+      type = boxType;
+    }
+};
+
+GridBox parkingLot[PARKING_ROWS][PARKING_COLS] = {
+    //00                01             02
+    {GridBox(Occupied), GridBox(Path), GridBox(Occupied)},
+    //10                11             12
+    {GridBox(Unoccupied), GridBox(Path), GridBox(Occupied)},
+    //20                21             22
+    {GridBox(Occupied), GridBox(Path), GridBox(Occupied)},
+    //30                31             32
+    {GridBox(Occupied), GridBox(Path), GridBox(Occupied)},
+    //40                41             42
+    {GridBox(NaP),      GridBox(Start),GridBox(NaP)},
+};
```

Comment:
Comment would be good to add.

Also, would it be easy if we would like to scale our parking lot? And how would it look like if the parking lot was not rectangular?

Besides that, awesome work!

Reply:
Thanks for the suggestion, I have made the appropriate changes. Please let me know if the changes live up to your expectations.